client/httplib/httpclient.go

```diff
@@ -0,0 +1,296 @@
+// Copyright 2020 beego
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package httplib
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/beego/beego/v2/core/berror"
+)
+
+// Client provides an HTTP client supporting chain call
+type Client struct {
+	Name       string
+	Endpoint   string
+	CommonOpts []BeegoHttpRequestOption
+
+	Setting *BeegoHTTPSettings
+	pointer *responsePointer
```

Comment:
It's not thread-safe.

Reply:
There may be many goroutines use the same client. So you could not store `response` inside Client